src/poetry/console/commands/show.py

```diff
@@ -300,6 +315,12 @@ def _display_packages_information(
             color = "cyan"
             name = locked.pretty_name
             install_marker = ""
+
+            if self.option("top-level") and not any(
+                locked.is_same_package_as(r) for r in root.all_requires
```

Comment:
One more thing: Since `all_requires` is not an attribute but a property that iterates over groups and dependencies we should store it in a local variable outside of the loop so we have to calculate it just once. Further, I'd do the same for `self.option("top-level")`. It's not consistent for all options, but it's done for some options at least.